src/logging.cpp

```diff
@@ -296,10 +297,27 @@ void BCLog::Logger::LogPrintStr(const std::string& str)
                 setbuf(new_fileout, nullptr); // unbuffered
                 fclose(m_fileout);
                 m_fileout = new_fileout;
+
+                // Re-log important messages to the new file.
+                bool save_started_new_line = m_started_new_line;
+                m_started_new_line = true;
+                for (const std::string& msg : m_msgs_relog) {
+                    std::string s = AddPrefix(msg) + '\n';
+                    FileWriteStr(s, m_fileout);
+                    for (const auto& cb : m_print_callbacks) {
+                        cb(s);
+                    }
```

Comment:
It's bad that this loop is duplicated here, and weird too, because the various unknown future print callbacks probably don't care that the log was rotated. But this is necessary for the test (which uses the callback mechanism). It may be better for the test to simply open the real `debug.log` file, as it used to, even though that's not quite as nice and clean.